### PR TITLE
DM-7137: utils raDecStr delimiters are untested and don't work

### PR DESCRIPTION
**Pull request by jmeyers314**
I added tests that use the delimiters, and fixed/pythonized/deduplicated the code.

### CODE REVIEW

File: src/RaDecStr.cc [hunk 1]
```diff
@@ -121,111 +121,93 @@ string ut::decDegToStr(double decDeg) {
...
 //
 // Converting strings to numbers
 //
 
-double ut::raStrToRad(std::string raStr, std::string delimiter) {
-    return degToRad( raStrToDeg(raStr) );
-}
-
+double sexagesimalStrToDecimal(std::string inStr, std::string delimiter) {
+    double hh_or_dd, mm, ss;
```
Comment (timj) [inline src/RaDecStr.cc]: why were these moved to the top of the function?

**Comment by timj** [comment]
Do we understand why it wasn't working for me with non-standard delimiters?

File: src/RaDecStr.cc (hunk 1 above)
Comment (jmeyers314) [inline src/RaDecStr.cc]: Good point.  I'll declare down below where they first get used.

**Comment by jmeyers314** [comment]
>Do we understand why it wasn't working for me with non-standard delimiters?

I believe this was because the `boost::regex` object was initialized `static`, so it got set the first time through the function to use ':' as the delimiter (in the earlier tests), then ignored any request for a different delimiter when returning to the function in your new test.
